lib/commands/context.js

```diff
@@ -330,6 +330,9 @@ extensions.onPageChange = async function (pageChangeNotification) {
         await this.remote.pageLoad();
       }
 
+      logger.debug("Clearing any frames");
+      this.curWebFrames = [];
```

Comment:
seems reasonable to clear the frames if the page changes or reloads. should we not put this into the `if (needsPageLoad)` conditional above, though? otherwise we will clear the frames even when we don't think anything has happened

Reply:
The trouble is that I can't get it to go into that branch. And looking at the code closer I don't see how it could. `needsPageLoad` is "is `this.curContext` in exactly one of `this.contexts` and `pageArray`?", but we have synced those lists at line 277.

Reply:
> it could theoretically happen in between a user selecting a frame and trying to do something with it.

In a way that's exactly the problem. The user selects a frame and then tries to run some javascript in it, but in the meantime a new page has unexpectedly loaded under their feet, so the frame no longer exists.

Reply:
I see. So better for Appium not to think the frame exists and to return a NoSuchFrame error so the user can retry. Alright, I'm cool with this. I still want @imurchie's review though.

Reply:
I'm exploring this this morning.